pkg/maps/ctmap/ctmap.go

```diff
@@ -120,11 +120,22 @@ type CtMapRecord struct {
 
 // InitMapInfo builds the information about different CT maps for the
 // combination of L3/L4 protocols.
-func InitMapInfo(registry *metrics.Registry, v4, v6, natRequired bool) {
-	global4Map, global6Map := nat.GlobalMaps(registry, v4, v6, natRequired)
+func InitMapInfo(registry *metrics.Registry, v4, v6 bool, nat4 nat.NatMap4, nat6 nat.NatMap6) {
+	var global4Map, global6Map *nat.Map
 	global4MapLock := &lock.Mutex{}
 	global6MapLock := &lock.Mutex{}
 
+	if nat4 != nil {
```

Comment:
have you checked whether this nil check on an interface type actually works as expected. we have a lot of cases where this doesn't work as expected (and we fallback to a reflection based nil check)